pombola/kenya/management/commands/kenya_check_for_recurring_problems.py

```diff
@@ -0,0 +1,152 @@
+    Organisation, Position, PositionTitle, approximate_date_to_date)
+
+def admin_url(o):
+    return urljoin('http://info.mzalendo.com', o.get_admin_url())
+
+
+def existing_overlapping_party_position(position, correct_title):
+    """Check if there is already an overlapping party position"""
+    party = position.organisation
+    assert party.kind.name == 'Political Party'
+    person = position.person
+    earliest_start_date = approximate_date_to_date(
+        position.start_date, 'earliest')
+    latest_end_date = approximate_date_to_date(
+        position.end_date, 'latest')
+    for existing in Position.objects.filter(
+            person=person,
+            title=correct_title,
+            organisation=party,
+    ):
+        if existing.approximate_date_overlap(
+                earliest_start_date,
+                latest_end_date):
+            return True
+    return False
+
+
+class ProblemFromQuerySet(object):
+
+    """A class to represent simple types of problem to fix"""
+
+    def __init__(self, qs, fix_kwargs):
+        self.qs = qs
+        self.model = qs.model
+        self.fix_kwargs = fix_kwargs
+
+    def count(self):
+        return self.qs.count()
+
+    def itermodels(self):
+        for o in self.qs:
+            yield o
+
+    def fix_problems(self):
+        return self.qs.update(**self.fix_kwargs)
+
+
+class ParliamentPositionProblem(ProblemFromQuerySet):
+
+    """A class for fixing Kenyan parliamentary organisation errors"""
+
+    def __init__(self, qs, parliament_organisation):
+        super(ParliamentPositionProblem, self).__init__(qs, None)
+        self.parliament_organisation = parliament_organisation
+
+    def fix_problems(self):
+        fixed = 0
+        member_pt = PositionTitle.objects.get(name='Member')
+        for p in self.qs:
+            print "  Fixing position", p
+            # There are really two cases here; one where the position
+            # just represents the membership of parliament, and one
+            # where it represents the party that they were a member of
+            # while a member of parliament. We want to make sure that
+            # both of the positions are still represented afterwards.
+            # (In the future these will hopefully both be represented by
+            # single Position with an equivalent of Popolo memberships'
+            # on_behalf_of property.)
+            org = p.organisation
+            org_kind = org.kind
+            if org_kind.name == 'Political Party':
+                if not existing_overlapping_party_position(p, member_pt):
+                    new_position = Position.objects.create(
+                        person=p.person,
+                        organisation=org,
+                        start_date=p.start_date,
+                        end_date=p.end_date,
+                        title=member_pt)
+                    print "    Creating a party position for the same period:", new_position
```

Comment:
I'm correct in my understanding of this that it will take cases where the position has a title of "Member of the National Assembly" but the organisation is a party when it should be Parliament. And that the creation bit is to make sure we don't remove any party membership data while we correct this?